storm-core/src/jvm/org/apache/storm/trident/operation/impl/JoinerMultiReducer.java

```diff
@@ -139,21 +138,5 @@ private boolean increment(List[] lengths, int[] indices, int j, int overrideInde
         }
         return true;
     }
-    
-    public static class JoinState {
-        List<List>[] sides;
-        int numSidesReceived = 0;
-        int[] indices;
-        TridentTuple group;
-        
-        public JoinState(int numSides, TridentTuple group) {
-            sides = new List[numSides];
-            indices = new int[numSides];
-            this.group = group;
-            for(int i=0; i<numSides; i++) {
-                sides[i] = new ArrayList<List>();
-            }            
-        }
-    }
-    
+
```

Comment:
Moved JoinState to its own Java file since two classes refer JoinState now.
It's a kind of public class, but only JoinerMultiReducer uses this and it resides on `impl` directory so I think it's OK to move.